src/main/java/com/ably/kafka/connect/ChannelSinkTask.java

```diff
@@ -46,17 +46,18 @@
     public void start(Map<String, String> settings) {
         logger.info("Starting Ably channel Sink task");
 
-        final ChannelSinkConnectorConfig config = new ChannelSinkConnectorConfig(settings);
+        final ChannelSinkConnectorConfig connectorConfig = new ChannelSinkConnectorConfig(settings);
         final ConfigValueEvaluator configValueEvaluator = new ConfigValueEvaluator();
-        channelSinkMapping = new DefaultChannelSinkMapping(config, configValueEvaluator);
-        messageSinkMapping = new MessageSinkMappingImpl(config, configValueEvaluator);
+        final ChannelConfig channelConfig = new ChannelConfigImpl(connectorConfig);
+        channelSinkMapping = new DefaultChannelSinkMapping(connectorConfig, configValueEvaluator, channelConfig);
+        messageSinkMapping = new MessageSinkMappingImpl(connectorConfig, configValueEvaluator);
 
-        if (config.clientOptions == null) {
+        if (connectorConfig.clientOptions == null) {
             logger.error("Ably client options were not initialized due to invalid configuration.");
```

Comment:
In passing, recognising that this is not code you modified within the scope of this pull request, this error message is misleading.

Reply:
Fair enough. 🙂 Leave it as is, it just seemed a bit vague.